### PR TITLE
Added users to federation

### PR DESCRIPTION
**Pull request by Skarlso**
## Summary

Adds `users` to Federation.

Related to #4.

### CODE REVIEW

**Comment by Skarlso** [comment]
something wrong with the linter again :/

**Comment by yitsushi** [comment]
```
  Error: 1-60 lines are duplicate of `services/federation/users_test.go:1-61` (dupl)
  Error: 1-61 lines are duplicate of `services/federation/instances_test.go:1-60` (dupl)
```

**Comment by yitsushi** [comment]
rebase from `main`, i added a golangci configuration file to ignore dupl (and a few others) in tests.

File: services/federation/users_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package federation_test
+
+import (
+	"log"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
```
Comment (yitsushi) [inline services/federation/users_test.go:10]: this line is the last error with `gci` error.

**Comment by yitsushi** [comment]
The linter works as intended (ignore all the tar errors, not an issue, hope the will fix it, but it's there even if the PR passes the check. Scroll down and check teh results instead of all the tar errors.

And you can always run the linter locally.

**Comment by yitsushi** [comment]
Please update the README too as this PR resolves #4

**Review by yitsushi — APPROVED** [review]

**Pull request merged**